winch/filetests/filetests/x64/br/as_call_all.wat

```diff
@@ -11,11 +11,11 @@
 ;;    8:	 897c2414             	mov	dword ptr [rsp + 0x14], edi
 ;;    c:	 89742410             	mov	dword ptr [rsp + 0x10], esi
 ;;   10:	 8954240c             	mov	dword ptr [rsp + 0xc], edx
-;;   14:	 4c89742404           	mov	qword ptr [rsp + 4], r14
-;;   19:	 b8ffffffff           	mov	eax, 0xffffffff
-;;   1e:	 4883c418             	add	rsp, 0x18
-;;   22:	 5d                   	pop	rbp
-;;   23:	 c3                   	ret	
+;;   14:	 4c893424             	mov	qword ptr [rsp], r14
+;;   18:	 b8ffffffff           	mov	eax, 0xffffffff
+;;   1d:	 4883c418             	add	rsp, 0x18
+;;   21:	 5d                   	pop	rbp
+;;   22:	 c3                   	ret	
```

Comment:
Is this an expected/intended change in this first commit? I thought this was purely a refactoring to prep for multi-value, and I wouldn't expect any functional changes. Certainly not ones where we are changing from where we are pushing values to the stack when I don't see any corresponding changes in how we pop values from the stack.

But maybe this is addressed in follow up commits that I haven't gotten to yet...

Reply:
It's indeed intended, I just forgot to mention it in the commit message: I realized that the slot for the `VMContext` was not aligned, so I introduced alignment for it, similar to what I did for the return pointer. 

In the first commit, that is here https://github.com/bytecodealliance/wasmtime/pull/7535/commits/5dcefe38b9dc4cac3f24bf9f4d0eb20f43f9b6a1#diff-b396908c6cc30abb2ad12e93c9443d5b8875dd320980cde6896068f5e5cbf4c1R101-R106; hopefully this clarifies things!

Reply:
That makes sense -- thanks for the explanation!